cranelift/codegen/src/regalloc/solver.rs

```diff
@@ -1149,24 +1149,14 @@ mod tests {
     use super::{Move, Solver};
     use crate::entity::EntityRef;
     use crate::ir::Value;
-    use crate::isa::{RegClass, RegInfo, RegUnit, TargetIsa};
+    use crate::isa::registers::{RegBank, RegClassData};
+    use crate::isa::{RegClass, RegInfo, RegUnit};
     use crate::regalloc::RegisterSet;
-    use alloc::boxed::Box;
-    use core::str::FromStr;
-    use target_lexicon::triple;
+    use core::borrow::Borrow;
 
-    // Make an arm32 `TargetIsa`, if possible.
-    fn arm32() -> Option<Box<dyn TargetIsa>> {
-        use crate::isa;
-        use crate::settings;
-
-        let shared_builder = settings::builder();
-        let shared_flags = settings::Flags::new(shared_builder);
-
-        isa::lookup(triple!("arm"))
-            .ok()
-            .map(|b| b.finish(shared_flags))
-    }
+    // Arm32 `TargetIsa` is now `TargetIsaAdapter`, which does not hold any info
+    // about registers, so we directly access `INFO` from registers-arm32.rs.
+    include!(concat!(env!("OUT_DIR"), "/registers-arm32.rs"));
```

Comment:
Maybe use x86 or riscv instead?

Reply:
I believe some of the tests here or in regalloc/pressure.rs use the fact that S, D and Q registers overlap. I am not sure if we can achieve something similar with x86 or risc-v registers.